packages/jest-worker/src/base/BaseWorkerPool.ts

```diff
@@ -85,15 +90,33 @@ export default class BaseWorkerPool {
     throw Error('Missing method createWorker in WorkerPool');
   }
 
-  end(): void {
+  async end(): Promise<PoolExitResult> {
```

Comment:
I'm not sure if this should be considered a breaking change or not (thus should only land in 25).

(I know I asked you to make it async, but changing the return type to be a promise, async fn or not, is a change in signature)

25 shouldn't be far out, thoughts on holding off?

Reply:
TBH I'm not sure about the change from throw to reject you asked me to make.
The argument was that it's annoying if a function can both throw and reject.
I think that this function should *only ever throw*, never reject. It has the case of invalid usage (`end()` called multiple times), but it should never "fail to end" the pool. If there's no other way, `SIGKILL` everything. It should always be able to clean up.
If we'd revert that back then it's definitely not breaking because the workers were always `end`ed asynchronously, there was just no way to find out when that happened (and the signature itself is backward compatible).

Reply:
@SimenB would you be fine with changing this back? I think calling this with invalid preconditions should throw immediately and not worst case even result in an uncaught rejection